src/libraries/System.Transactions.Local/src/System/Transactions/Oletx/DTCTransactionManager.cs

```diff
@@ -0,0 +1,151 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Transactions.Diagnostics;
+
+namespace System.Transactions.Oletx
+{
+    internal sealed class DtcTransactionManager
+    {
+        private string? _nodeName;
+        private OletxTransactionManager _oletxTm;
+        private IDtcProxyShimFactory _proxyShimFactory;
+        private byte[] _whereabouts = null!; // Late-initialized
```

Comment:
Nit: Seems like this should really be byte[]?, given all the places null is assigned to this.  The Whereabouts property can be the only place that needs to understand and cast-away the maybe-nullness as needed.

Related, I'm curious why the _initialized flag is needed at all... can this being null serve the same purpose as _initialized == false?

Reply:
Makes sense, doing both.